### DISCUSSION TITLE
Suggestion for Sniffer dongle with tshark  - linux machine

### DISCUSSION THREAD

**Discussion by jchellap** [openthread/openthread#9408] [General]
Hi All,

Can someone suggest any sniffer dongle that can work in linux platform using wireshark/tshark?
Any suggestion will help.

Thanks

**Comment by jwhui** [comment]
Any OpenThread RCP should support packet sniffing with Wireshark:
- [Packet Sniffing using Extcap](https://openthread.io/guides/pyspinel/sniffer-extcap).

IEEE 802.15.4 vendors also provide their own solutions:
- [nRF Sniffer for 802.15.4](https://www.nordicsemi.com/Products/Development-tools/nrf-sniffer-for-802154)

If others have solutions they'd like to share, please do so!

**Reply by jchellap** [reply]
Thanks @jwhui 
I want in linux platform. I see tshark is not supporting nRF. So want to know in linux platform have used sniffers.
If yes, let me know so i can get more details on the issue I am facing and take help to fix it.

**Reply by jwhui** [reply]
Both options I provided above support Linux. I suggest clicking through the links and reading the documentation.

**Reply by jchellap** [reply]
Hi @jwhui 
I have referred this in the past. What issue i am observing is the wireshark or tshark doesnt list the nRF interfaces when i give
wireshark -D or tshark -D
Also, the dongle is enumerated as /dev/ttyACM0
extcap for nRF is used in windows within wireshark directory. I am not sure for linux where the file should be copied.
I have given permissions to open the port as well.
My linux machine uses root  mode. Any idea does it really affect?
Let me know if i am missing something in configuring. I will revisit the link and look for more info

**Comment by Chapoly1305** [comment]
Thank you Hui, the direct links saved lots of my time. 
the sniffer plug-in is tested working with nrf52840 as well, Wireshark 4.2.2 on Ubuntu 22.04.
